build_all.sh

```diff
@@ -28,8 +28,10 @@ cd ./build
 
 CMAKE_FLAGS="-DCMAKE_INSTALL_PREFIX=../ -DEMC_EXEC_DIR=ON"
 
-if [[ "$target" != "wcoss_cray" && "$target" != "odin" ]]; then
-  CMAKE_FLAGS+=" -DCMAKE_Fortran_COMPILER=ifort -DCMAKE_C_COMPILER=icc"
+if [[ "$compiler" == "intel" ]]; then
```

Comment:
Why is this entire block necessary at all? The compilers should be set correctly by loading the modules, not by adding duct tape inside the build scripts.

Reply:
So, in the build module do: setenv CMAKE_Fortran_COMPILER ifort

Reply:
For the sake of this PR moving ahead, shall we just keep it as it is?

Reply:
I can remove this block and test the compile on Hera, Orion, Jet, WCOSS-Cray, WCOSS-Dell.

Reply:
If I remove the entire block, the build will default to the GNU compiler (on the machines I have access to).  So additional changes will be necessary.  But if I just remove the first 'if' block ("$compiler == "intel") it works correctly.

Reply:
> If I remove the entire block, the build will default to the GNU compiler (on the machines I have access to). So additional changes will be necessary. But if I just remove the first 'if' block ("$compiler == "intel") it works correctly.

Thanks for trying. We need to keep the outer intel test until we fix the cmake build, otherwise it would break the build for someone trying to use the GNU compilers on one of these platforms. I think the PR is good to go in its current form.

Reply:
> > If I remove the entire block, the build will default to the GNU compiler (on the machines I have access to). So additional changes will be necessary. But if I just remove the first 'if' block ("$compiler == "intel") it works correctly.
> 
> Thanks for trying. We need to keep the outer intel test until we fix the cmake build, otherwise it would break the build for someone trying to use the GNU compilers on one of these platforms. I think the PR is good to go in its current form.

If the outer 'if' block is kept, I might add something like "compiler=${compiler:-intel}".  I did not know to set 'compiler' and the script errored at line 31.

Reply:
I see - do you want me to make this change?

Reply:
> I see - do you want me to make this change?

Please.  That would help the user.  The default does not have to be intel.  You could also do something like "compiler=${compiler:?}".   Anything to alert the user to set 'compiler'.

Reply:
@DomHeinzeller, @GeorgeGayno-NOAA, I just merged this PR, but then noticed this comment wasn't resolved.  Does this change still need to be made?  Thanks.

Reply:
Not sure, I lost track because this has been sitting there for so long. But the PR hasn't been approved by @GeorgeGayno-NOAA or @LarissaReames-NOAA and better should not have been merged?